assign_rights/assemble.py

```diff
@@ -20,11 +21,14 @@ def get_date_value(self, object, field_name, request_date, period):
         Args:
             object: the object for which a date is to be calculated.
             field_name: the object attribute containing date data.
-            request_date: date data sumbitted in a request.
+            request_date: representation of a date sumbitted in a request.
             period: the numbe of years to be used in calculating the date.
+
+        Returns:
+            A date object representation of the date after calculation.
         """
         if not getattr(object, field_name):
-            return request_date + relativedelta(years=period)
+            return datetime.strptime(request_date, "%Y-%m-%d").date() + relativedelta(years=period)
```

Comment:
OK, this is fine for now, but let's remember that we're asking for a specific type of date formatting, which we need to check for somewhere.

Reply:
Also, how about using `date.fromisoformat()` ? https://docs.python.org/3/library/datetime.html#datetime.date.fromisoformat

Reply:
@helrond @bonniegee As referenced in the latest commit, I'm using `date.fromisoformat()`, but that requires Python 3.7, so I've also updated the Dockerfile.

Reply:
@p-galligan Ooh, I missed that detail! I believe we are running Python 3.6 on our servers, so we may need to go back to the previous implementation: 

```
datetime.strptime(request_date, "%Y-%m-%d").date()
```

Sorry about the confusion! That will teach me to read the documentation carefully...